SIL.Core/Reflection/ReflectionHelper.cs

```diff
@@ -461,13 +463,21 @@ public static string LongVersionNumberString
 			}
 		}
 
+		private static readonly bool RunningFromUnitTest = AppDomain.CurrentDomain.GetAssemblies()
+			.Any(assem => 
+				assem.FullName.StartsWith("nunit.framework", StringComparison.OrdinalIgnoreCase)
+				|| assem.FullName.StartsWith("Microsoft.VisualStudio.QualityTools.UnitTestFramework", StringComparison.OrdinalIgnoreCase)
+				|| assem.FullName.StartsWith("Microsoft.VisualStudio.TestPlatform.TestFramework", StringComparison.OrdinalIgnoreCase)
+				|| assem.FullName.StartsWith("xunit.core", StringComparison.OrdinalIgnoreCase)
+			);
+
 		public static string DirectoryOfTheApplicationExecutable
 		{
 			get
 			{
 				string path;
-				bool unitTesting = Assembly.GetEntryAssembly() == null;
-				if (unitTesting)
+				// checking for assembly == null is for backwards compatibility with old code that may unintentionally call take the unit test path
```

Comment:
I think there is a typo (or some other source of unintelligible grammer) in this comment.